cpp/CMakeLists.txt

```diff
@@ -49,16 +75,33 @@ foreach(proto_file ${proto_files})
     # Add generated files to the list
     list(APPEND generated_files ${out_proto_src} ${out_proto_hdr} ${out_grpc_src} ${out_grpc_hdr})
 
-    # Generated sources
-    add_custom_command(
-        OUTPUT "${out_proto_src}" "${out_proto_hdr}" "${out_grpc_src}" "${out_grpc_hdr}"
-        COMMAND ${_PROTOBUF_PROTOC}
-        ARGS --grpc_out "${OUT_FOLDER}"
-            --cpp_out "${OUT_FOLDER}"
-            -I "${PROTO_FOLDER}"
-            --plugin=protoc-gen-grpc="${_GRPC_CPP_PLUGIN_EXECUTABLE}"
-            "${proto_file}"
-      DEPENDS "${proto_file}")
+    # When using protoc from FetchContent the protobuf  source have to be included manually
+    if(USE_SYSTEM_GRPC)
+        add_custom_command(
+            OUTPUT "${out_proto_src}" "${out_proto_hdr}" "${out_grpc_src}" "${out_grpc_hdr}"
+            COMMAND ${_PROTOBUF_PROTOC}
+            ARGS --grpc_out "${OUT_FOLDER}"
+                --cpp_out "${OUT_FOLDER}"
+                -I "${PROTO_FOLDER}"
+                --plugin=protoc-gen-grpc="${_GRPC_CPP_PLUGIN_EXECUTABLE}"
+                "${proto_file}"
+          DEPENDS "${proto_file}")
+    else()
+        set(PROTOBUF_INCLUDE_FOLDER "${CMAKE_CURRENT_LIST_DIR}/cmake/build/_deps/grpc-src/third_party/protobuf/src")
```

Comment:
Do you not need to add it for both?

I couldn't find an example in the CPP gRPC example directory to confirm it, [but they don't use the Google types there](https://github.com/search?q=repo%3Agrpc%2Fgrpc+google++path%3Aexamples%2Fcpp+lang%3A%22Protocol+Buffers%22+&type=code).